Dockerfile

```diff
@@ -1,46 +1,55 @@
 
-# Install the Agent
-RUN echo "deb http://apt.datadoghq.com/ stable main" > /etc/apt/sources.list.d/datadog.list \
- && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys C7A7DA52 \
- && apt-get update \
- && apt-get install --no-install-recommends -y datadog-agent="${AGENT_VERSION}" \
- && apt-get clean \
- && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+# prevent the agent from being started after install
+ENV DD_START_AGENT 0
+
+# Add Docker check
+COPY conf.d/docker_daemon.yaml "$DD_HOME/agent/conf.d/docker_daemon.yaml"
+
+# Add install and config files
+ADD https://raw.githubusercontent.com/DataDog/dd-agent/master/packaging/datadog-agent/source/setup_agent.sh /setup_agent.sh
+COPY entrypoint.sh /entrypoint.sh
+
+# Install minimal dependencies
+RUN apk add --update curl-dev python-dev tar sysstat
+
+# Install optional dependencies
+RUN apk add gcc musl-dev pgcluster-dev linux-headers
+
+# Expose supervisor port
+EXPOSE 9001/tcp
+
+# Expose DogStatsD port
+EXPOSE 8125/udp
+
+# Install the agent
+RUN sh /setup_agent.sh
 
 # Configure the Agent
 # 1. Listen to statsd from other containers
 # 2. Turn syslog off
 # 3. Remove dd-agent user from supervisor configuration
-# 4. Remove dd-agent user from init.d configuration
-# 5. Fix permission on /etc/init.d/datadog-agent
-# 6. Remove network check
-# 7. Symlink Dogstatsd to allow standalone execution
-RUN mv /etc/dd-agent/datadog.conf.example /etc/dd-agent/datadog.conf \
- && sed -i -e"s/^.*non_local_traffic:.*$/non_local_traffic: yes/" /etc/dd-agent/datadog.conf \
- && sed -i -e"s/^.*log_to_syslog:.*$/log_to_syslog: no/" /etc/dd-agent/datadog.conf \
- && sed -i "/user=dd-agent/d" /etc/dd-agent/supervisor.conf \
- && sed -i 's/AGENTUSER="dd-agent"/AGENTUSER="root"/g' /etc/init.d/datadog-agent \
- && chmod +x /etc/init.d/datadog-agent \
- && rm /etc/dd-agent/conf.d/network.yaml.default \
- && ln -s /opt/datadog-agent/agent/dogstatsd.py /usr/bin/dogstatsd
-
-# Add Docker check
-COPY conf.d/docker_daemon.yaml /etc/dd-agent/conf.d/docker_daemon.yaml
-
-COPY entrypoint.sh /entrypoint.sh
+# 4. Remove network check
+# 5. Symlink Dogstatsd to allow standalone execution
+# 6. Remove setup script
+RUN cp "$DD_HOME/agent/datadog.conf.example" "$DD_HOME/agent/datadog.conf" \
+  && sed -i -e"s/^.*non_local_traffic:.*$/non_local_traffic: yes/" "$DD_HOME/agent/datadog.conf" \
+  && sed -i -e"s/^.*log_to_syslog:.*$/log_to_syslog: no/" "$DD_HOME/agent/datadog.conf" \
+  && sed -i "/user=dd-agent/d" "$DD_HOME/agent/supervisor.conf" \
+  && rm "$DD_HOME/agent/conf.d/network.yaml.default" \
+  && ln -s /opt/datadog-agent/agent/dogstatsd.py /usr/bin/dogstatsd \
+  && rm /setup_agent.sh
 
 # Extra conf.d and checks.d
 VOLUME ["/conf.d"]
 VOLUME ["/checks.d"]
 
-# Expose DogStatsD port
-EXPOSE 8125/udp
-
 ENTRYPOINT ["/entrypoint.sh"]
-CMD ["supervisord", "-n", "-c", "/etc/dd-agent/supervisor.conf"]
+
+CMD cd "$DD_HOME" && source venv/bin/activate && supervisord -c agent/supervisor.conf
```

Comment:
1. why not set `WORKDIR` instead of `cd`?
 1. also, if you need to chain commands, i suggest you call `cmd.sh`, less error prone